### PR TITLE
[PACT-811] Implemented local message removal, after processing.

### PR DESCRIPTION
**Pull request by tna-erasmos**
Also added tests for `LocalMessageStore`, as well as introducing `Version1UUID`.

### CODE REVIEW

**Review by felixgb — COMMENTED** [review]
Just a few small comments.

File: src/main/scala/uk/gov/nationalarchives/omega/api/common/Version1UUID.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+sealed trait Version1UUID {
+  val value: UUID
+}
+
+object Version1UUID {
+
+  private val generator = Generators.timeBasedGenerator(EthernetAddress.fromInterface)
+
+  private class Version1UUIDImpl(override val value: UUID) extends Version1UUID {
```
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply src/main/scala/uk/gov/nationalarchives/omega/api/common/Version1UUID.scala:36]: Sure. As long we can still protect the constructor for an invalid UUID.

Looking into it ...

File: src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalMessageStore.scala [hunk 2]
```diff
@@ -22,38 +22,41 @@
...
+        case Failure(e) =>
+          logger.error(s"Failed to persist the message for ID [$messageId]")
+          Failure[Version1UUID](e)
       }
     }
-  }
 
-  private def writeFile(path: Path, messageText: String): IO[Unit] =
+  def removeMessage(messageId: Version1UUID): IO[Try[Unit]] =
+    if (messageId != null) removeFile(generateFilePath(messageId))
```
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalMessageStore.scala:53]: Looking into it ...

[the remark this answers is not in the document]
Comment (tna-erasmos) [reply src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalMessageStore.scala:53]: I'll remove the check and the related spec.

After all, we're not doing it elsewhere in code base.

File: src/main/scala/uk/gov/nationalarchives/omega/api/common/Version1UUID.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply src/main/scala/uk/gov/nationalarchives/omega/api/common/Version1UUID.scala:36]: Easy. Will change.

File: src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalMessageStore.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalMessageStore.scala]: Changed.

**Review by felixgb — APPROVED** [review]

**Review by rwalpole — APPROVED** [review]

**Pull request merged**